src/planning/plan/plan-consumer.ts

```diff
@@ -10,14 +10,15 @@
 
 import {assert} from '../../platform/assert-web.js';
 import {Arc} from '../../runtime/arc.js';
+import {Consumer} from '../../runtime/hot.js';
 import {SuggestionComposer} from '../suggestion-composer.js';
 
 import {PlanningResult} from './planning-result.js';
 import {Suggestion, SuggestionVisibilityOptions} from './suggestion.js';
 import {SuggestFilter} from './suggest-filter.js';
 import {PlannerInspector} from '../planner-inspector.js';
 
-type Callback = ({}) => void;
```

Comment:
Can we move this type of a centralized place within `plan/`, and re-use the same `Callback` type where `Consumer<{}>` is used? 

I think the scope should be everything in the `plan` or `planning` package, since these seems to be a module wide pattern.